lib/elixir/lib/task.ex

```diff
@@ -255,4 +267,131 @@ defmodule Task do
+
+  In case the task process dies, this function will exit with the
+  same reason as the task.
+
+  This function assumes the task's monitor is still active or the monitor's
+  `:DOWN` message is in the message queue. If it has been demonitored, or the
+  message already received, this function wait for the duration of the timeout
+  awaiting the message.
+  """
+  @spec yield(t, timeout) :: {:ok, term} | nil
+  def yield(%Task{ref: ref} = task, timeout \\ 5_000) do
+    receive do
+      {^ref, reply} ->
+        Process.demonitor(ref, [:flush])
+        {:ok, reply}
+      {:DOWN, ^ref, _, _, reason} ->
+        exit({reason(reason, task), {__MODULE__, :yield, [task, timeout]}})
+    after
+      timeout ->
+        nil
+    end
+  end
+
+  @doc """
+  Unlink and shutdown the task then check for a reply.
+
+  Returns `{:ok, reply}` if the reply is received while shutting down the task,
+  otherwise `nil`.
+
+  The shutdown method is either a timeout or `:brutal_kill`. In the case
+  of a `timeout`, a `:shutdown` exit signal is sent to the task process
+  and if it does not exit within the timeout it is killed. With `:brutal_kill`
+  the task is killed straight away. In the case that the task exits abnormal,
+  or a timeout shutdown kills the task, this function will exit with the same
+  reason.
+
+  It is not required to call this function when terminating the caller, unless
+  exiting with reason `:normal` or the task is trapping exits. If the caller is
+  exiting with a reason other than `:normal` and the task is not trapping exits the
+  caller's exit signal will stop the task. The caller can exit with reason
+  `:shutdown` to shutdown linked processes, such as tasks, that are not trapping
+  exits without generating any log messages.
+
+  This function assumes the task's monitor is still active or the monitor's
+  `:DOWN` message is in the message queue. If it has been demonitored, or the
+  message already received, this function will block forever awaiting the message.
+  """
+  @spec shutdown(t, timeout | :brutal_kill) :: {:ok, term} | nil
+  def shutdown(task, shutdown \\ 5_000)
+
+  def shutdown(%Task{pid: pid} = task, :brutal_kill) do
+    exit(pid, :kill)
+    case shutdown_receive(task, :brutal_kill, :infinity) do
+      {:error, reason} ->
+        exit({reason, {__MODULE__, :shutdown, [task, :brutal_kill]}})
+      result ->
+        result
+    end
+  end
+
+  def shutdown(%Task{pid: pid} = task, timeout) do
+    Process.exit(pid, :shutdown)
+    case shutdown_receive(task, :shutdown, timeout) do
+      {:error, reason} ->
+        exit({reason, {__MODULE__, :shutdown, [task, timeout]}})
+      result ->
+        result
+    end
+  end
+
+  ## Helpers
+
+  defp reason(:noconnection, %{pid: pid}), do: {:nodedown, node(pid)}
+  defp reason(reason, _),                  do: reason
+
+  # spawn a process to ensure task gets exit signal if process dies from exit signal
+  # between unlink and exit.
+  defp exit(task, reason) do
```

Comment:
Or that or we have a process around just to do this. I guess the process is fine?